commands/ping.go

```diff
@@ -35,29 +45,33 @@ trip latency information.
 
 		numPings, _ := req.Options["count"].(uint)
 
-		ch, err := GetAPI(env).Ping().Ping(req.Context, peerID, numPings, time.Second)
+		pingCh, err := GetPorcelainAPI(env).NetworkPing(req.Context, peerID)
 		if err != nil {
 			return err
 		}
 
-		for p := range ch {
-			if err := re.Emit(p); err != nil {
+		for i := uint(0); numPings == 0 || i < numPings; i++ {
+			pong, pingChOpen := <-pingCh
+			result := &PingResult{
+				Count: i,
+				Time:  pong,
+			}
+			if err := re.Emit(result); err != nil {
 				return err
 			}
+			if !pingChOpen {
+				return errors.New("Ping channel closed by receiver")
+			}
 		}
+
 		return nil
 	},
 	Encoders: cmds.EncoderMap{
-		cmds.Text: cmds.MakeTypedEncoder(func(req *cmds.Request, w io.Writer, p *api.PingResult) error {
-			if len(p.Text) > 0 {
-				fmt.Fprintln(w, p.Text) // nolint: errcheck
-			} else if p.Success {
-				fmt.Fprintf(w, "Pong received: time=%.2f ms\n", p.Time.Seconds()*1000) // nolint: errcheck
-			} else {
-				fmt.Fprintf(w, "Pong failed\n") // nolint: errcheck
-			}
+		cmds.Text: cmds.MakeTypedEncoder(func(req *cmds.Request, w io.Writer, result *PingResult) error {
+			milliseconds := result.Time.Seconds() * 1000
+			fmt.Fprintf(w, "Pong received: seq=%d time=%.2f ms\n", result.Count, milliseconds) // nolint: errcheck
```

Comment:
might be easier to understand if we just print result.Time.Round(time.Millisecond).String()? at the very least fractions of ms dont matter in this context, ms is sufficient